python-threatexchange/threatexchange/signal_type/signal_base.py

```diff
@@ -143,29 +148,31 @@ class TextHasher(FileHasher):
     """
 
     @classmethod
+    @abc.abstractmethod
     def hash_from_str(cls, text: str) -> str:
         """
         Get a string representation of the hash from a string.
 
         If a hash cannot be generated, empty string should be returned.
         """
-        raise NotImplementedError
+        pass
 
     @classmethod
     def hash_from_file(cls, file: pathlib.Path) -> str:
         return cls.hash_from_str(file.read_text())
 
 
-class MatchesStr:
+class MatchesStr(abc.ABC):
     @classmethod
+    @abc.abstractmethod
     def matches_str(cls, signal: str, haystack: str) -> SignalComparisonResult:
         """
         Compare the distance of two hashes, the key operation for matching.
```

Comment:
nit/ignorable: naming is hard, it a little confusing that this doc string is the exact same as they one used for `compare_hash` might be worth updating one / clarifying the difference